opensrp-giz-malawi/src/main/java/org/smartregister/giz/presenter/GizMaternityProfileActivityPresenter.java

```diff
@@ -0,0 +1,26 @@
+package org.smartregister.giz.presenter;
+
+import org.smartregister.anc.library.util.ConstantsUtils;
+import org.smartregister.commonregistry.CommonPersonObjectClient;
+import org.smartregister.giz.application.GizMalawiApplication;
+import org.smartregister.maternity.activity.BaseMaternityProfileActivity;
+import org.smartregister.maternity.contract.MaternityProfileActivityContract;
+import org.smartregister.maternity.presenter.MaternityProfileActivityPresenter;
+import org.smartregister.maternity.utils.MaternityConstants;
+
+public class GizMaternityProfileActivityPresenter extends MaternityProfileActivityPresenter {
+
+    public GizMaternityProfileActivityPresenter(MaternityProfileActivityContract.View profileView) {
+        super(profileView);
+    }
+
+    @Override
+    public boolean hasAncProfile() {
+        BaseMaternityProfileActivity activity = (BaseMaternityProfileActivity) getProfileView();
+        CommonPersonObjectClient commonPersonObjectClient = (CommonPersonObjectClient) activity.getIntent()
+                .getSerializableExtra(MaternityConstants.IntentKey.CLIENT_OBJECT);
+        return GizMalawiApplication.getInstance().gizEventRepository()
```

Comment:
This should be called from a different thread to prevent not-responsive UI. Can you confirm this?

Reply:
at the moment its not will update this. Good catch